File: Python/面试经典150题/数组、目录/80_删除有序数组中的重复项Ⅱ.py
from typing import List
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        index, pre = 1, 0
        for i in range(1, len(nums)):
            if nums[i] != nums[i-1]:
                nums[index] = nums[i]
                index += 1
                pre = 0
            else:
                pre += 1
                if pre == 1:
                    nums[index] = nums[i]
                    index += 1
        return index

nums = [0,0,1,1,1,1,2,3,3]
print(Solution().removeDuplicates(nums))

File: JS/面试经典150题/数组、字符串/80_删除有序数组中的重复项Ⅱ.js
var removeDuplicates = function(nums) {
    let gap = 0, count=0
    for(let i=1; i<nums.length; i++){
        if(nums[i] === nums[i-1]){
            count++
            if(count > 1)
                gap += 1
            nums[i - gap] = nums[i]
        }
        else{
            nums[i - gap] = nums[i]
            count = 0
        }
    }
    return nums.length - gap
};

nums = [1,1,1,1,2,2,3]
console.log(removeDuplicates(nums));
console.log(nums);